src/object/srv_obj_migrate.c

```diff
@@ -1370,22 +1356,19 @@ migrate_fetch_update_bulk(struct migrate_one *mrone, daos_handle_t oh,
 	 * this data shard.
 	 */
 
-	if (mrone->mo_iods_num_from_parity > 0) {
-		daos_epoch_t min_eph = DAOS_EPOCH_MAX;
+	for (i = 0; i < mrone->mo_iods_num_from_parity; i++) {
+		for (j = 0; j < mrone->mo_iods_from_parity[i].iod_nr; j++) {
+			daos_iod_t iod = mrone->mo_iods_from_parity[i];
 
-		for (i = 0; i < mrone->mo_iods_num_from_parity; i++) {
-			for (j = 0; j < mrone->mo_iods_from_parity[i].iod_nr; j++)
-				min_eph = min(min_eph,
-					      mrone->mo_iods_update_ephs_from_parity[i][j]);
+			iod.iod_nr = 1;
+			iod.iod_recxs = &mrone->mo_iods_from_parity[i].iod_recxs[j];
+			rc = __migrate_fetch_update_bulk(mrone, oh, &iod, 1,
```

Comment:
just confirm that now the parity exts will be rebuild one by one, as previous code the take the min epoch to update multiple parity ext may cause client-side degraded fetch's parity epoch mismatch issue?

Reply:
The same question

Reply:
yes, especially if the shard needs to be relocated to different shard due to exclude/reint etc, and if there are no available shards to reconstruct the shard, so it has to fetch from the original shard, then we have to rebuild one by one, and can not merge the shard, which might ruin the data.